torch/testing/_asserts.py

```diff
@@ -102,6 +102,71 @@ def wrapper(actual: Tensor, expected: Tensor, **kwargs: Any) -> Optional[Excepti
     return wrapper
 
 
+def _check_sparse_coo_members_individually(
```

Comment:
Do we want to include nnz in this? Just in case someone changes it to, for example, include the 0s in the values and not treat it as a count of sparse (I've seen this called "unspecified") entries?

Reply:
> Do we want to include nnz in this? Just in case someone changes it to, for example, include the 0s in the values and not treat it as a count of sparse (I've seen this called "unspecified") entries?

@cpuhrsch I don't object to including nnz in the check but I think there is no point: nnz is not really a member of a coo tensor that the user can specify incorrectly. nnz is a computed value (and is equal to the numel of values) and hence there is nothing to check.

I think it is very unlikely that someone will change the semantics of nnz. nnz has the meaning of representing the number of specified values, even when some of these values can actually be specified as zeros, and only this meaning makes sense in the implementations of various operations.

Btw, https://github.com/pytorch/pytorch/pull/45695 proposes the rename nnz -> nse.

Reply:
From a call with @pmeier , introducing nnz check makes sense for coalesced tensors in which case it is possible to have a pair of (mathematically equal) tensors with the same shape but different nnz.

Reply:
Actually a `._nnz()` check is useful in case we check uncoalesced tensors. They can have the same `.shape`, but different `._nnz()`. In such a case our current implementation would try to check `._indices()` of different shapes for equality, which would result in an uncaught exception. If we simply require `._nnz()` to match before we check indices and values, we can circumvent this problem.